src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshalling.cs

```diff
@@ -337,47 +337,63 @@ void AnalyzeSignatureType(ImmutableArray<Location> locations, ITypeSymbol type)
 
             private bool TypeIsAutoLayoutOrContainsAutoLayout(ITypeSymbol type)
             {
-                Debug.Assert(type.IsValueType);
-
-                if (_isAutoLayoutOrContainsAutoLayoutCache.TryGetValue(type, out bool isAutoLayoutOrContainsAutoLayout))
+                return TypeIsAutoLayoutOrContainsAutoLayout(type, ImmutableHashSet<ITypeSymbol>.Empty.WithComparer(SymbolEqualityComparer.Default));
```

Comment:
Just out of curiosity: If we just passed in type.OriginalDefintion here, does it fix the issue without requiring any other changes below?

Reply:
We still need the rest of the changes for the “recursive struct definition” scenario so we don’t stack overflow on invalid code.